doc/conf.py

```diff
@@ -17,6 +17,9 @@
 import matplotlib
 import sphinx
 
+# This is only necessary to monkey patch the signature later on.
+from sphinx_gallery import gen_rst
```

Comment:
This should be moved below the `_check_deps` call  below to prevent `ImportError`s when sphinx-gallery is not yet installed.

Reply:
I made a new PR #11221 for that. Thanks for taking a close look.

Reply:
thanks